tests/scripts/thread-cert/Cert_5_3_06_RouterIdMask.py

```diff
@@ -81,14 +91,56 @@ def test(self):
         time.sleep(5)
         self.assertEqual(self.nodes[ROUTER2].get_state(), 'router')
 
+        leader_messages = self.sniffer.get_messages_sent_by(DUT_LEADER)
+        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
+        self.assertTrue(command.check_id_set(msg, self.nodes[ROUTER2]))
+        self.assertNotEqual(command.check_routing_cost(msg, self.nodes[ROUTER2]), 0)
+
+        # 2
         self.nodes[ROUTER2].reset()
         self._setUpRouter2()
 
-        time.sleep(300)
-
+        # 3 & 4
+        # Test if the cost from LEADER to ROUTER2 is infinity, verify it appear in 12 mins.
+        flag_in_id_set = False
+        routing_cost = 1
+        for i in range(0, 24):
+            time.sleep(30)
+            leader_messages = self.sniffer.get_messages_sent_by(DUT_LEADER)
+            msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT, False)
+            if msg == None:
+                continue
+
+            flag_in_id_set = command.check_id_set(msg, self.nodes[ROUTER2])
+            routing_cost = command.check_routing_cost(msg, self.nodes[ROUTER2])
+            if not flag_in_id_set or (flag_in_id_set and routing_cost == 0):
```

Comment:
From the coding's perspective, above line is equivalent with 
`if not flag_in_id_set or routing_cost == 0:`

Reply:
yes, it's equivalent. But I think it is much easier to understand.